File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02696/p02696_num3_parsed.py
import math

def max_difference(A, B, N):
    max_val = -float('inf')
    for x in range(N+1):
        difference = math.floor(A*x/B) - A*math.floor(x/B)
        max_val = max(max_val, difference)
    return max_val

A, B, N = map(int, input().split())
result = max_difference(A, B, N)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02696/p02696_num3_parsed.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int A, B, N;
    cin >> A >> B >> N;

    int maxVal = 0;
    for (int x = 0; x <= N; x++) {
        maxVal = max(maxVal, (A * x / B) - A * (x / B));
    }

    cout << maxVal << endl;

    return 0;
}